src/main/java/com/ltbc/swarmrestapi/service/SwarmRestApiService.java

```diff
@@ -0,0 +1,82 @@
+package com.ltbc.swarmrestapi.service;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.ltbc.swarmrestapi.service.beesinfo.BeesInfoDetails;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Service;
+
+import java.io.IOException;
+import java.net.URI;
+import java.net.http.HttpClient;
+import java.net.http.HttpRequest;
+import java.net.http.HttpResponse;
+
+@Service
+public class SwarmRestApiService {
+
+    private static final String OPENAI_URL = "https://api.openai.com/v1/chat/completions";
+
+    @Value("${OPENAI_API_KEY}")
+    private String apiKey;
+    HttpClient client = HttpClient.newHttpClient();
+
+    private static final Logger logger = LoggerFactory.getLogger(SwarmRestApiService.class);
+
+    public BeesInfoDetails sendPromptToOpenAi(String requestPrompt) throws IOException, InterruptedException {
+        if (apiKey == null) {
+            return null; // TODO: fix up these to return an empty BeesInfoDetails obj
```

Comment:
@lindentree I think we randomly considered using an API key per user (to offload the cost to the end user). Did you get the chance to look into that? (cc @Clue355 @idjevm)

Reply:
I think we’d have to set up a client side thing for the user to enter their chatgpt key and authorize its usage for our third party app.

Reply:
@lindentree Let's bring it up with Moni and Shamique :) I'll post on Slack